website/server/controllers/api-v4/auth.js

```diff
@@ -84,8 +86,10 @@ api.verifyUsername = {
 
     const issues = verifyUsername(req.body.username, res);
 
-    const count = await User.count({ 'auth.local.lowerCaseUsername': req.body.username.toLowerCase() });
-    if (count > 0)  issues.push(res.t('usernameTaken'));
+    const existingUser = await User.findOne({ 'auth.local.lowerCaseUsername': req.body.username.toLowerCase() }, {auth: 1}).exec();
+    if (existingUser !== undefined && existingUser !== null && existingUser._id !== user._id) {
```

Comment:
For these types of existence checks if (existingUser) works like if (existingUser !== undefined && existingUser !== null)